packages/devtools_app/lib/src/memory/flutter/memory_snapshot_models.dart

```diff
@@ -0,0 +1,657 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:vm_service/vm_service.dart';
+
+import '../../trees.dart';
+import '../flutter/memory_controller.dart';
+import '../flutter/memory_graph_model.dart';
+
+/// Consolidated list of libraries.  External is the external heap
+/// and Filtered is the sum of all filtered (hidden) libraries.
+const String externalLibraryName = 'External';
+const String filteredLibrariesName = 'Filtered';
+
+/// Name for predefined Reference and FieldReference should never be
+/// seen by user.
+const String emptyName = '<empty>';
+const String sentinelName = '<sentinel>';
+
+class Reference extends TreeNode<Reference> {
+  Reference._empty()
+      : controller = null,
+        name = emptyName,
+        isLibrary = false,
+        isExternal = false,
+        isFiltered = false,
+        isClass = false,
+        isObject = false,
+        actualClass = null;
+
+  Reference._sentinel()
+      : controller = null,
+        name = sentinelName,
+        isLibrary = false,
+        isExternal = false,
+        isFiltered = false,
+        isClass = false,
+        isObject = false,
+        actualClass = null;
+
+  Reference.createLibrary(this.controller, this.name,
```

Comment:
nit. make this
`Reference.library` instead of `Reference.createLibrary` same for the others to follow dart conventions for constructor names.

Reply:
Only issue was with createClass since class was a reserved name which is why I used createXXX.  But I changed everything and createClass is not aClass,